### DISCUSSION TITLE
colorprofile

### DISCUSSION THREAD

**Discussion by aroehrin** [ImageMagick/ImageMagick#6469] [Digital Image Processing]
A question about the usage of colorprofiles and the expected results. I have some ECI RGB files which were transformed with photoshop to CMYK using a PSO coated icc profile. The transformation wasn't done by me. Now I try to reproduce the results with imagemagick and by adding the option "-profile" and the corresponding PSO coated icc file.
The results look different and with a colorpicker tool I can see that different numbers were created.
I tried different options (e.g. using the "-strip" parameter) and I received different results but never a match with the transformation of photoshop.
I have limited knowledge about the usage of colorprofiles. What is the expectation with transformation using colorprofiles and different tools like imagemagick and photoshop? Should the results be identical or does it depend on internal algorithms?
If the results should be identical: what needs to be considered in imagemagick to get the same  results?
Thanks for any hints

**Comment by snibgo** [comment]
When converting from one ICC colour profile to another, the action of `-profile` is modified by `-black-point-compensation` and `-intent`. If those settings are the same as in Photoshop, I would expect the results to be similar.

They will not be identical. First, because "intents" are not well-defined, and different software can use different algorithms. (ImageMagick uses LCMS, and is likely to make the same results as other software that uses LCMS. I don't suppose Photoshop uses LCMS.) Second, because of arithmetic rounding errors.
